libyul/backends/evm/OptimizedEVMCodeTransform.cpp

```diff
@@ -69,35 +69,39 @@ void OptimizedEVMCodeTransform::operator()(CFG::FunctionCall const& _call)
 	// Validate stack.
 	{
 		yulAssert(m_assembly.stackHeight() == static_cast<int>(m_stack.size()), "");
-		yulAssert(m_stack.size() >= _call.function.get().arguments.size() + 1, "");
+		yulAssert(m_stack.size() >= _call.function.get().arguments.size() + (_call.canContinue ? 1 : 0), "");
 		// Assert that we got the correct arguments on stack for the call.
 		for (auto&& [arg, slot]: ranges::zip_view(
 			_call.functionCall.get().arguments | ranges::views::reverse,
 			m_stack | ranges::views::take_last(_call.functionCall.get().arguments.size())
 		))
 			validateSlot(slot, arg);
 		// Assert that we got the correct return label on stack.
-		auto const* returnLabelSlot = get_if<FunctionCallReturnLabelSlot>(
-			&m_stack.at(m_stack.size() - _call.functionCall.get().arguments.size() - 1)
-		);
-		yulAssert(returnLabelSlot && &returnLabelSlot->call.get() == &_call.functionCall.get(), "");
+		if (_call.canContinue)
+		{
+			auto const* returnLabelSlot = get_if<FunctionCallReturnLabelSlot>(
+				&m_stack.at(m_stack.size() - _call.functionCall.get().arguments.size() - 1)
+			);
+			yulAssert(returnLabelSlot && &returnLabelSlot->call.get() == &_call.functionCall.get(), "");
+		}
```

Comment:
Generally, the ``OptimizedEVMCodeTransform`` only ensures that each of the operations for which it generates code has everything it needs on stack, while the rest of the stack be arbitrary (giving the ``StackLayoutGenerator`` the freedom to do whatever it wants). Conceptually, the layout transition to the entry layout of the next operation would just remove the return label slot, if it was there (well, in the cases here, since the function will revert, there won't really be a "next operation", but still :-)).

So, in practice, there will never be a return label slot in the opposite case, so we *could* assert this here, but I'd still keep the ``OptimizedEVMCodeTransform`` agnostic to the stack layouts below the current operation it's looking at.